### PR TITLE
added better invitations info, endpoints for getting accepted invitat…

### PR DESCRIPTION
**Pull request by DFSko**
…ions list, endpoint for disable invitation

### CODE REVIEW

File: Api/Services/Agents/AgentInvitationService.cs [hunk 1]
```diff
@@ -62,12 +63,12 @@ public Task<Result<AgentInvitationInfo>> GetPendingInvitation(string invitationC
             => _invitationService.GetPendingInvitation<AgentInvitationInfo>(invitationCode, UserInvitationTypes.Agent);
 
 
-        public Task<List<AgentInvitationResponse>> GetAgentInvitations(int agentId)
-            => GetInvitations(i => i.Data.AgentId == agentId);
+        public Task<List<AgentInvitationResponse>> GetAgentInvitations(int agentId, bool isAccepted)
```
Comment (hasmikharutyunyan) [inline Api/Services/Agents/AgentInvitationService.cs:66]: maybe will be understandable to have two methods ?

File: HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace HappyTravel.Edo.Data.Migrations
+{
+    public partial class AddIsDisabledToUserInvitations : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<bool>(
+                name: "IsDisabled",
```
Comment (hasmikharutyunyan) [inline HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs:10]: We are using `isActive` name in other places, maybe here also will be better to use that name for consistency ?

File: Api/Controllers/AgentControllers/AgentsController.cs [hunk 3]
```diff
@@ -181,30 +181,73 @@ public async Task<IActionResult> GetInvitationData(string code)
...
 
 
         /// <summary>
-        ///    Gets agent invitations list
+        ///    Gets agent not accepted invitations list
         /// </summary>
         [HttpGet("agent/invitations")]
         [ProducesResponseType(typeof(List<AgentInvitationResponse>), (int) HttpStatusCode.OK)]
-        public async Task<ActionResult<List<AgentInvitationResponse>>> GetOwnInvitations()
+        public async Task<ActionResult<List<AgentInvitationResponse>>> GetOwnNotAcceptedInvitations()
```
Comment (hasmikharutyunyan) [inline Api/Controllers/AgentControllers/AgentsController.cs:222]: maybe better to use `agent` instead ow `own` ?

File: Api/Models/Agents/AgentInvitationResponse.cs [hunk 4]
```diff
@@ -6,14 +6,16 @@ namespace HappyTravel.Edo.Api.Models.Agents
...
-            string position, string email)
+            string position, string email, string inviting, string created)
         {
             Id = id;
             Title = title;
             FirstName = firstName;
             LastName = lastName;
             Position = position;
             Email = email;
+            Inviting = inviting;
```
[the remark this answers is not in the document]
Comment (mingazhev) [reply Api/Models/Agents/AgentInvitationResponse.cs:17]: I think this is a bit tautologic. Maybe something like CreatedBy?

File: Api/Services/Users/UserInvitationService.cs [hunk 5]
```diff
@@ -101,23 +99,54 @@ public Task<Result<TInvitationData>> GetPendingInvitation<TInvitationData>(strin
...
+                    .SingleOrDefaultAsync(i => i.CodeHash == invitationCode);
+
+                return invitation ?? Result.Failure<InvitationBase>("Invitation not found");
+            }
+
+
+            async Task DisableInvitation(InvitationBase invitation)
+            {
+                invitation.IsDisabled = true;
+                _context.Entry(invitation).Property(i => i.IsDisabled).IsModified = true;
```
[the remark this answers is not in the document]
Comment (DFSko) [reply Api/Services/Users/UserInvitationService.cs:146]: context.Update updating all properties, we only need one

File: Api/Services/Agents/AgentInvitationService.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Agents/AgentInvitationService.cs]: fixed

File: Api/Controllers/AgentControllers/AgentsController.cs (hunk 3 above)
Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentsController.cs]: fixed

File: Api/Models/Agents/AgentInvitationResponse.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (DFSko) [reply Api/Models/Agents/AgentInvitationResponse.cs]: fixed

File: HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs (hunk 2 above)
Comment (DFSko) [reply HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs:10]: If use `IsActive`, we need to set all invitations as active, but default boolean value is false

Comment (hasmikharutyunyan) [reply HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs:10]: We can change default in migration, but I do not insist.

File: HappyTravel.Edo.Data/Agents/InvitationBase.cs [hunk 6]
```diff
@@ -9,6 +9,7 @@ public abstract class InvitationBase
         public string Email { get; set; }
         public DateTime Created { get; set; }
         public bool IsAccepted { get; set; }
+        public bool IsDisabled { get; set; }
```
Comment (kirillta) [inline HappyTravel.Edo.Data/Agents/InvitationBase.cs:12]: Not clear, What `disabled` means?

File: HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs (hunk 2 above)
Comment (kirillta) [reply HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs:10]: Let's change

Comment (DFSko) [reply HappyTravel.Edo.Data/Migrations/20210111074209_AddIsDisabledToUserInvitations.cs]: fixed

File: HappyTravel.Edo.Data/Agents/InvitationBase.cs (hunk 6 above)
Comment (DFSko) [reply HappyTravel.Edo.Data/Agents/InvitationBase.cs]: Changed to IsActive

**Review by mingazhev — APPROVED** [review]

**Review by hasmikharutyunyan — APPROVED** [review]

**Pull request merged**